docker/docker-compose.yml

```diff
@@ -0,0 +1,52 @@
+version: '3'
+services:
+  snp_automate_2023:
+    build:
+      context: ..
+      dockerfile: docker/Dockerfile
+      args:
+        - TAG=humble-master
+    environment:
+      DISPLAY: $DISPLAY
+      XAUTHORITY: $XAUTHORITY
+      # NVIDIA_DRIVER_CAPABILITIES: all
+      ROS_LOG_DIR: /tmp/.ros
+      ROS_LOCALHOST_ONLY: $ROS_LOCALHOST_ONLY
+      ROS_DOMAIN_ID: $ROS_DOMAIN_ID
+    container_name: snp_automate_2023
+    image: ghcr.io/ros-industrial-consortium/snp_automate_2023:humble-master
+    stdin_open: true
+    tty: true
+    network_mode: host
+    privileged: true
+    user: ${CURRENT_UID}  # CURRENT_UID=$(id -u):$(id -g)
+    group_add:
+      - sudo
+    volumes:
+      - /tmp/.X11-unix:/tmp/.X11-unix
+      - /etc/hosts:/etc/hosts
+      - /etc/passwd:/etc/passwd:ro
+      - /etc/group:/etc/group:ro
+      - /etc/shadow:/etc/shadow:ro
+      - /dev:/dev
+    # deploy:
+    #   resources:
+    #     reservations:
+    #       devices:
+    #         - driver: nvidia
+    #           count: 1
+    #           capabilities: [gpu]
+  # Driver
+  micro-ros-agent:
+    environment:
+      ROS_LOG_DIR: /tmp
+      ROS_LOCALHOST_ONLY: $ROS_LOCALHOST_ONLY
+      ROS_DOMAIN_ID: $ROS_DOMAIN_ID
+    container_name: micro-ros-agent
+    image: microros/micro-ros-agent:humble
+    command: udp4 --port 8888
+    stdin_open: true
+    tty: true
+    privileged: true
```

Comment:
is this really needed, or copy-pasta from the service definition above?

Reply:
I believe line 47 is needed since [motoros2 specifies that command](https://github.com/Yaskawa-Global/motoros2?tab=readme-ov-file#using-docker-linux-only) for `docker run`. I am not certain on the rest. @marip8 do we need lines 48-50?

Reply:
47 to 49 are ok. I was mostly asking about `priviliged`.

(GitHub highlights more lines than I selected here)

Reply:
To my understanding, `privileged` makes your user effectively equivalent to `root`. I don't think this should be necessary since we've configured everything in the Docker image to be accessible by "regular" named users.

Reply:
The Agent certainly doesn't need it. That was mostly why I commented on it.

Reply:
I see; I didn't notice this was in the agent section. Definitely agree, and I don't think the SNP container needs privileged mode